src/main/java/gregtech/client/model/OreBakedModel.java

```diff
@@ -0,0 +1,226 @@
+                return getOverlayQuads(side, rand);
+            } else {
+                return Collections.emptyList();
+            }
+        }
+
+        protected List<BakedQuad> getOverlayQuads(@Nullable EnumFacing side, long rand) {
+            int index = side == null ? 6 : side.getIndex();
+            if (this.overlayQuads[index] == null) {
+                List<BakedQuad> quads = new ArrayList<>(this.overlay.getQuads(null, side, rand));
+                for (int i = 0; i < quads.size(); i++) {
+                    quads.set(i, transform(quads.get(i)));
+                }
+                return this.overlayQuads[index] = quads;
+            }
+            return this.overlayQuads[index];
+        }
+
+        private static BakedQuad transform(BakedQuad quad) {
+            if (FMLClientHandler.instance().hasOptifine()) return quad;
+            VertexFormat format = quad.getFormat();
+            if (!format.getElements().contains(DefaultVertexFormats.TEX_2S)) {
+                format = new VertexFormat(quad.getFormat());
+                format.addElement(DefaultVertexFormats.TEX_2S);
+            }
+            UnpackedBakedQuad.Builder builder = new UnpackedBakedQuad.Builder(format) {
+                @Override
+                public void put(int element, float... data) {
+                    if (this.getVertexFormat().getElement(element) == DefaultVertexFormats.TEX_2S)
+                        super.put(element, 480.0f / 0xFFFF, 480.0f / 0xFFFF);
+                    else super.put(element, data);
+                }
+            };
+            quad.pipe(builder);
+            builder.setApplyDiffuseLighting(false);
+            return builder.build();
+        }
+    }
+
+    private static final class Entry {
+
+        private final StoneType stoneType;
+        private final MaterialIconSet iconSet;
+        private final boolean emissive;
+
+        private Entry(StoneType stoneType, MaterialIconSet iconSet, boolean emissive) {
+            this.stoneType = stoneType;
+            this.iconSet = iconSet;
+            this.emissive = emissive;
+        }
+
+        public ModelResourceLocation getModelId() {
+            return new ModelResourceLocation(new ResourceLocation(GTValues.MODID,
+                    "ore_" + this.stoneType.name + "_" + this.iconSet.name + (this.emissive ? "_emissive" : "")), "");
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Entry entry = (Entry) o;
+            return this.stoneType.name.equals(entry.stoneType.name) &&
+                    this.iconSet.name.equals(entry.iconSet.name) &&
+                    this.emissive == entry.emissive;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(stoneType.name, iconSet.name, emissive);
```

Comment:
Given that all of the instance variables are final, this hash can be pre-computed in the constructor in order to prevent creating objects (the varargs array for Objects#hash) in hashCode().